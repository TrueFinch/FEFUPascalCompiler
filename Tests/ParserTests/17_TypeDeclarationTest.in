program a;
type rec = record
            i: integer;
           end;
           
     int = integer;
     
     RecArr15 = array [0 .. 15] of rec;
var
    i: int;
    arr: RecArr15;
begin
end.