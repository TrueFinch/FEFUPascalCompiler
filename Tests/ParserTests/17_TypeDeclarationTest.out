                                                  a                                                    
                                                  ┼                                                    
                                              MainBlock                                                
                          ┌───────────────────────┴────────────────┬────────────────────────┐          
                        type                                      var               CompoundStatement  
        ┌──────────────┬──┴──────────────┐                  ┌──────┴───────┐                           
    TypeDecl        TypeDecl          TypeDecl        SimpleVarDecl  SimpleVarDecl                     
  ┌─────┴─┐        ┌───┴──┐        ┌─────┴────┐       ┌───┬─┴         ┌────┴─┐                         
 rec  RecordType  int  integer  recarr15  ArrayType   i  int         arr  intarr15                     
          ┼                                 ┌─┴───┐                                                    
          :                                ..    rec                                                   
      ┌───┴─┐                             ┌─┴┐                                                         
      i  integer                          0  15                                                        